docs/winrt/WinRT SxS Components.md

```diff
@@ -1 +1,153 @@
-## Placeholder
\ No newline at end of file
+# Side-by-Side Versioned Components
+
+## Status
+Proposed: This document describes a proposed design that has not been implemented.
+
+## Background
+The Windows Runtime type system was largely designed with the needs of opertating system components in mind. Although applications can define their own API types using the Windows Runtime type system, the feature set for doing so is somewhat basic. Versioning strategies are only vaguely defined, and no provisions are made for side-by-side support of a component. As the community has developed more sophisticated libraries over time, these deficiencies are becoming more challenging, and further support is needed.
+
+Today, the Windows Runtime only supports binary-compatible versioning. That is, if a an application or component is developed with a dependency on version N of another component, version N+1 *can* be developed in such a way that code designed to use version N can leverage version N+1 without changes or recompilation of the client component. Alternately, a component author can ignore binary compatbility (which comes with trade-offs) and require that applications use a single, specific version of the component. 
+
+Up to this point, this has been largely sufficient. The single largest "component" is the Windows operating system, which has processes in place to ensure binary compatbility. Other components have typically been distributed as part of an MSIX packaged application. he expectation is that the application developer can manage the dependency graph. This has largely been true up to this point. However, authors of new components are recognizing that they will be consumed in more complex dependency graphs, where the application may have dependencies on multiple versions of the component, and the component author does not want to be locked into a binary-stable model. This implies the need for some form of side by side support.
+
+## Terminology
+* Cross-talk - any ambiguity or contention caused by the presence of two highly similar pieces of code that are not intended to interact, such as two versions of the same component.
```

Comment:
Nit on markdown formatting - I think you need a blank line here for some previewers to work right.

Reply:
I'm using a markdown analyzer (and now a spell checker too), but it didn't flag that. Do you have a favorite that you prefer (ideally for vscode)?

Reply:
For VSCode, I just use "Prettier - Code Formatter" which would have also converted the \* into \-, so YMMV.